PWGHF/TableProducer/HFTreeCreatorLbToLcPi.cxx

```diff
@@ -0,0 +1,220 @@
+               soa::Join<aod::McParticles, aod::HfCandLbMCGen> const& particles,
+               aod::BigTracksPID const& tracks)
+  {
+
+    // Filling event properties
+    rowCandidateFullEvents.reserve(collisions.size());
+    for (auto& collision : collisions) {
+      rowCandidateFullEvents(
+        collision.bcId(),
+        collision.numContrib(),
+        collision.posX(),
+        collision.posY(),
+        collision.posZ(),
+        0,
+        1);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
+      auto fillTable = [&](int CandFlag,
+                           int FunctionSelection,
+                           float FunctionInvMass,
+                           float FunctionCt,
+                           float FunctionY) {
+        if (FunctionSelection >= 1) { // Set to true to keep unselected events as well
+          rowCandidateFull(
+            candidate.rSecondaryVertex(),
+            candidate.decayLength(),
+            candidate.decayLengthXY(),
+            candidate.decayLengthNormalised(),
+            candidate.decayLengthXYNormalised(),
+            candidate.chi2PCA(),
+            candidate.impactParameterNormalised0(),
+            candidate.ptProng0(),
+            RecoDecay::P(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0()),
+            candidate.impactParameterNormalised1(),
+            candidate.ptProng1(),
+            RecoDecay::P(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1()),
+            candidate.pxProng0(),
+            candidate.pyProng0(),
+            candidate.pzProng0(),
+            candidate.pxProng1(),
+            candidate.pyProng1(),
+            candidate.pzProng1(),
+            candidate.impactParameter0(),
+            candidate.impactParameter1(),
+            candidate.errorImpactParameter0(),
+            candidate.errorImpactParameter1(),
+            1 << CandFlag,
+            FunctionInvMass,
+            candidate.pt(),
+            candidate.p(),
+            candidate.cpa(),
+            candidate.cpaXY(),
+            FunctionCt,
+            candidate.eta(),
+            candidate.phi(),
+            FunctionY,
+            candidate.flagMCMatchRec());
+        }
+      };
+
+      fillTable(0, candidate.isSelLbToLcPi(), InvMassLbToLcPi(candidate), CtLb(candidate), YLb(candidate));
+    }
+
+    // Filling particle properties
+    rowCandidateFullParticles.reserve(particles.size());
+    for (auto& particle : particles) {
+      if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::LbToLcPi) {
+        rowCandidateFullParticles(
+          particle.mcCollision().bcId(),
+          particle.pt(),
+          particle.eta(),
+          particle.phi(),
+          RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode())),
+          particle.flagMCMatchGen());
```

Comment:
Why do you need `flagMCMatchGen`?

Reply:
I use it to split signal and background events when training the BDT

Reply:
But here you select only signal so the flag always has the same value.

Reply:
Correct, the " if (FunctionSelection >= 1) {" seemed to be the default in other tree creators, so I kept that. However, when I use it generate signal and background trees, I set it to 'if(true)'

Reply:
`FunctionSelection` is irrelevant here. This is a loop over generated particles which you selected with
```c++
if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::LbToLcPi) {
```
so what you store are only Lb and anti-Lb.

Reply:
I am so sorry, I was looking at the wrong lines. The loop over particles was copied from the other tree creators and I actually don't use it whatsoever. For the sake of uniformity, it was left as-is. Would you suggest to remove it until needed?